### ISSUE TITLE
Combining external login provider won't work with ASP.NET Core Identity

### ISSUE THREAD

**Issue by g0dpain** [aspnet-contrib/AspNet.Security.OAuth.Providers#701]
I am using the discord provider which on its own functions well. 
Adding Identity to the application removes the ability of the discord callback, since it seems ``app.UseAuthentication()`` to override something.
But removing it from the Startup Identity and signing in with identity's ``SignInManager<TUser>`` stops working.

Since I am new to this whole OAuth thing, is this behaviour intended?

**Comment by martincostello** [comment]
I'm not aware of this particular behaviour, so I'm not sure what could be wrong.

If it helps you, this repository uses ASP.NET Core Identity with this repo's Amazon, Apple and GitHub providers and works as expected (at least how I expect it to 😄): https://github.com/martincostello/alexa-london-travel-site

Maybe clone it locally and compare how it's set up compared to your own code?

**Comment by g0dpain** [comment]
@martincostello Thanks for your response, I will definitely look into it!

**Comment by g0dpain** [comment]
Sadly I couldn't figure out why it does not work. I even tried in a seperate new project to grab components of yours and it didn't.

I just created a project https://github.com/g0dpain/DiscordOAuth2.Test.git 
Do you maybe have time to look over it? I can also provide some client secrets and connection string via email for debugging.

**Comment by martincostello** [comment]
Sorry, I don't really have enough free time at the moment to look into issues outside the scope of this repo.

**Comment by g0dpain** [comment]
Thats alright.
After I rewrote the whole authentication thing I found out that I got authorized and signed in, although the callback path (in my case: /signin-discord) wouldn't get called even if I added it as an option ``discordOptions.CallbackPath = new PathString("/signin-discord")``

**Comment by g0dpain** [comment]
Well, got it working now.

Then there was a problem with getting the external login info, which I found multiple devs that shared the same problem in closed issues here.
``GetExternalLoginInfoAsync()`` returned always null
This was because this method tries to find the corresponding provider which was not set.
Adding ``var authProperties = _signInManager.ConfigureExternalAuthenticationProperties(provider, "/auth/complete");`` and providing it to ``ChallengeResult`` in the HTTP-Post sign-in method fixed it.
Changing the redirectUri in ``AuthenticationProperties`` fixed the problem not getting redirected to my ``signin-discord`` route with ``app.UseAuthentication()`` too.

I overlooked this simple line in your repo..
It maybe would be a great help for others to add this information to the projects sample
